autode/wrappers/G09.py

```diff
@@ -50,15 +50,26 @@
 
     for keyword in calc_input.keywords.copy():
 
-        # Add any empirical dispersion
-        if isinstance(keyword, kws.DispersionCorrection):
-            new_keywords.append(f'EmpiricalDispersion={keyword.g09}')
+        # Replace the basis set file specification with genecp
+        if str(keyword).endswith('.gbs'):
+            logger.info('Found a custom basis set file adding genecp')
+            new_keywords.append('genecp')
+            continue
+
+        if isinstance(keyword, kws.Keyword):
+            kwd_str = keyword.g09 if hasattr(keyword, 'g09') else keyword.g16
```

Comment:
This seems dodgy, is the assumption here that g16 keywords won't break things, they just might not do anything?

Reply:
agree that it's ugly. The trouble being this function is called for both g09 and g16 calculations but there's no access to which method. I think the below means there has to be either `.g09` or `.g16` if it's a keyword

```
if keyword.has_only_name():
        # set e.g. keyword.orca = 'b3lyp'
        setattr(keyword, self.method.name, keyword.name)
        continue
```

Open to suggestions though–

Reply:
ah yeah I see, this makes sense then